tests/main/systemd-success-exit-status/task.yaml

```
@@ -0,0 +1,87 @@
+summary: Check that services with success-exit-status have SuccessExitStatus directive
+
+details: |
+    Verifies that a snap service with SuccessExitStatus=42 250 is considered active or successfully terminated by systemd when it exits with status 42 or 250, and that other unexpected signals do not have this effect.
+
+environment:
+    SERVICE_NAME: snap.test-snapd-success-exit-status.test-snapd-success-exit-status.service
+
+restore: |
+    snap remove --purge test-snapd-success-exit-status || true
+
+execute: |
+    echo "Given a service snap is installed"
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-success-exit-status
+
+    echo "When the service state is reported as active"
+    while ! systemctl show -p ActiveState "$SERVICE_NAME" | grep -Pq "ActiveState=active"; do sleep 0.5; done
+
+    echo "The service contains the SuccessExitStatus= directive"
+    systemctl cat "$SERVICE_NAME" | grep -q "SuccessExitStatus=42 250"
```

We usually use `MATCH` instead of `grep -q` in spread tests

```suggestion
    systemctl cat "$SERVICE_NAME" | MATCH "SuccessExitStatus=42 250"
```